coil-core/src/nonAndroidMain/kotlin/coil3/Image.kt

```diff
@@ -44,7 +68,24 @@ class BitmapImage internal constructor(
     override val height: Int
         get() = bitmap.height
 
-    override fun asBitmap(): Bitmap {
-        return bitmap
+    override fun asPainter(): CoilPainter = CoilPainter.BitmapPainter {
+        bitmap
+    }
+}
+
+@ExperimentalCoilApi
+@Data
+class SvgImage internal constructor(
+    val svg: SVGDOM,
+    override val width: Int,
+    override val height: Int,
+): Image {
+    override val size: Long
+        get() = 4L * width * height
```

Comment:
The size of `SVGDOM` should be way smaller, but I didn't find a way to calculate it. I was thinking about setting the same size of the data coming from the `ImageSource`

Reply:
This should live in `coil-svg` as we can't have any SVG classes in the main artifact.